ShoppingCart-topic1/src/MailList.java

```diff
@@ -0,0 +1,35 @@
+import java.util.ArrayList;
+
+public class MailList implements List {
+
+	private ArrayList<String> notifications;
+	private static List instance = null;
+	private ArrayList<Manager> managers;
+
+	private MailList() {
+		notifications = new ArrayList<String>();
+		managers = new ArrayList<Manager>();
+	}
+
+	public static List getInstance() {
+		if (instance == null) {
+			instance = new MailList();
+		}
+		return instance;
+	}
+
+	public void addManager(Manager m) {
+		managers.add(m);
+	}
+
+	public void removeManager(Manager m) {
+		managers.remove(m);
+	}
+
+	public void notify(String not) {
+		notifications.add(not);
+		for (Manager m : managers) {
+			m.update(not);
+		}
+	}
+}

```

Comment:
I used the Observer pattern to modelate item 6. I choosed it because I have an object (MailList) who has to notify to managers(they are the observers) when something happen (new item/offer, etc).